### DISCUSSION TITLE
Cocalc example docker-compose.yml

### DISCUSSION THREAD

**Discussion by miohtama** [sagemathinc/cocalc#6710] [Show and tell]
Hi all,

As we are piloting the feasibility to replace Jupyter Hub with CoCalc,  [we have created an example `docker-compose.yml` to run CoCalc on-premise, or self-hosted](https://github.com/tradingstrategy-ai/cocalc-env).

The main driver is that we need to be able to have custom Python packages installed for the kernel. Futhermore, we need prefer to run on bare metal servers.

Some features

- Everything is orchestrated by single `docker-compose.yml` and command
- User Caddy-based HTTP basic auth to hide the instance behind username and password: this hides the server against all malicious robot traffic
- Shows how to create a new Docker imaged based on `cocalc-docker` version 2 and add more packages to it using `poetry`

The docker image is still not working perfectly, but the progress is promising.

**Comment by williamstein** [comment]
Very cool!   Please keep us posted, and let us know when you feel it is sufficiently stable, etc.,  so that we should also link to this from the main README.md. 

Also, if you want to have a video chat or something, let me know.

**Comment by miohtama** [comment]
We love open source and what CoCalc is doing. Happy to hop on a video chat after the initial hurdles and lessons learnt.
